model/flow/protocol_state.go

```diff
@@ -24,19 +24,60 @@ type DynamicIdentityEntryList []*DynamicIdentityEntry
 // plus some modifiers. We intend to restructure this code soon.
 // TODO: https://github.com/onflow/flow-go/issues/4649
 type ProtocolStateEntry struct {
-	// setup and commit event IDs for current epoch.
-	CurrentEpochEventIDs EventIDs
-	// setup and commit event IDs for previous epoch.
+	// Setup and commit event IDs for previous epoch.
 	PreviousEpochEventIDs EventIDs
-	// Part of identity table that can be changed during the epoch.
-	// Always sorted in canonical order.
-	Identities DynamicIdentityEntryList
+	// Protocol state for current epoch
+	CurrentEpoch EpochStateContainer
+	// Protocol state for next epoch. Could be nil if next epoch not setup yet.
+	NextEpoch *EpochStateContainer
 	// InvalidStateTransitionAttempted encodes whether an invalid state transition
 	// has been detected in this fork. When this happens, epoch fallback is triggered
 	// AFTER the fork is finalized.
 	InvalidStateTransitionAttempted bool
-	// NextEpochProtocolState describes protocol state of the next epoch
-	NextEpochProtocolState *ProtocolStateEntry
+}
+
+// EpochStateContainer is a container for data that is relevant to a single epoch.
+type EpochStateContainer struct {
+	// ID of setup event for this epoch, never nil.
+	SetupID Identifier
+	// ID of commit event for this epoch. Could be ZeroID if epoch was not committed.
+	CommitID Identifier
+	// Part of identity table that can be changed on a block-by-block basis.
+	// Each non-deferred identity-mutating operation is applied independently to each
+	// relevant epoch's dynamic identity list separately.
+	// Always sorted in canonical order.
+	Identities DynamicIdentityEntryList
```

Comment:
I would suggest to rename this field to `ActiveIdentities` and extend the godoc as follows:

```suggestion
	// ActiveIdentities contains the dynamic identity properties for the nodes that
	// are active in this epoch. Active means that these nodes are authorized to contribute
	// extending the chain. Nodes are listed in `ActiveIdentities` if and only if
	// they are part of the EpochSetup even for the respective epoch.
	// The dynamic identity properties can change from block to block. Each non-deferred
	// identity-mutating operation is applied independently to the `ActiveIdentities`
	// of the relevant epoch's EpochStateContainer separately.
	// ActiveIdentities are always sorted in canonical order.
	//
	// Context: In comparison, nodes that are joining in the next epoch or left as of this
	// epoch are only allowed to listen to the network but not actively contribute. Such
	// nodes are _not_ part of `ActiveIdentities`.
	ActiveIdentities DynamicIdentityEntryList
```